src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala

```diff
@@ -315,17 +333,128 @@ object AOIMonitorUtils extends Logging{
     }
   }
 
-  def getLastSequence(databaseURI: URI, procName: String)(implicit spark: SparkSession): Option[Int] = {
-    spark.sparkContext.parallelize(Seq(databaseURI)).map { uri =>
-      queryLastSequence(uri, procName)
-    }.collect.apply(0)
+  def setBeginSequence(interval: Interval, lastSequence: Int) = {
+    var connection: Connection = null
+    try {
+      connection = AOIDatabaseConfig.getConnection
+      val preppedStatement =
+        connection.prepareStatement("INSERT INTO checkpoint_interval VALUES (?, ?)")
+      preppedStatement.setString(1, interval.value)
+      preppedStatement.setLong(2, lastSequence)
+      preppedStatement.executeQuery()
+    } catch {
+      case e: Throwable => {
+        println(s"ERROR in setBeginSequence: ${e.getMessage}")
+      }
+    } finally {
+      if (connection != null) connection.close()
+    }
+  }
+
+  def queryAreasOfInterest(): List[AOIData] = {
+    var connection: Connection = null
+    val wkbReader = new WKBReader(new GeometryFactory(new PrecisionModel(), 4326))
+    val data = ListBuffer[AOIData]()
+    try {
+      connection = AOIDatabaseConfig.getConnection
+      val query =
+        """
+          |SELECT
+          |    aoi.id as aoi_id,
+          |    user_id,
+          |    aoi.created_at as created_at,
+          |    ST_AsBinary(geom) as geometry,
```

Comment:
This was orginally `ST_AsText` to pull the geoms as WKT, but WKB should be far more efficient.